docs/guides/transfer_learning.md

```diff
@@ -0,0 +1,165 @@
+---
+jupytext:
+  formats: ipynb,md:myst
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.13.8
+kernelspec:
+  display_name: 'Python 3.8.10 (''.venv'': venv)'
+  language: python
+  name: python3
+---
+
+# Transfer learning
+
++++
+
+This guide demonstrates how to perform transfer learning with Flax. There are various ways to do it, depending on your task. You can use a pretrained model as a feature extractor or fine-tune the entire model.
+
+You will learn how to do transfer learning for simple classification. The steps include how to:
+* Load a pretrained model from HuggingFace [Transformers](https://huggingface.co/docs/transformers/index).
+* Extract a Flax Module submodule from a pretrained model.
+* Create the classifier model.
+* Transfer the pretrained parameters to the new model structure.
+* Set up optimization for training different parts of the model separately with [Optax](https://optax.readthedocs.io/en/latest/).
+* Train the classifier.
+
+**Note:** Depending on your task, some of the content from this guide may be suboptimal. Specifically, if you are only going to train a linear classifier on top of a pretrained model, it may be better to just extract the feature embeddings once. This can result in much faster training, and you can use specialized algorithms for linear regression or logistic classification. This guide shows how to do end-to-end transfer learning with all the model parameters.
+
+## Setup
+
+```{code-cell} ipython3
+:tags: [skip-execution]
+
+! pip install transformers[flax]
+! pip install -U jax jaxlib
```

Comment:
And as discussed:

`!pip install -U flax jax jaxlib`, since `transformers` uses different versions of Flax and JAX.

Reply:
Let's remove the installation steps and have those steps https://github.com/google/flax/pull/2429. Note that `pip install optax` wasn't here.